components/homme/src/theta-l_kokkos/cxx/ForcingFunctor.hpp

```diff
@@ -461,24 +450,14 @@ class ForcingFunctor
 
       if (m_moist) {
         // Need to update dp, pnh and exner. Currently, pnh is storing pnh-pi
-        if (m_adjust_ps) {
-          Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
-                               [&](const int ilev) {
-            dp(ilev) = dp_adj(ilev);
-
-            pnh(ilev) += m_hvcoord.ps0*m_hvcoord.hybrid_am(ilev) +
-                                   ps *m_hvcoord.hybrid_bm(ilev);
-          });
-        } else {
-          // Compute hydrostatic p from dp. Store in exner, then add to pnh
-          auto p_i = Homme::subview(m_pi_i,kv.team_idx,igp,jgp);
-          m_elem_ops.compute_hydrostatic_p(kv,dp,p_i,exner);
-          Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
-                               [&](const int ilev) {
-            pnh(ilev) += exner(ilev);
-            dp(ilev) = dp_adj(ilev);
-          });
-        }
+        // Compute hydrostatic p from dp. Store in exner, then add to pnh
+        auto p_i = Homme::subview(m_pi_i,kv.team_idx,igp,jgp);
+        m_elem_ops.compute_hydrostatic_p(kv,dp_adj,p_i,exner);
+        Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
+                             [&](const int ilev) {
```

Comment:
This looks like a bug:  The variable "exner" is used to store pressure (this is a little confusing - we should add a comment noting this - I assume because we want to be spare in our use of GPU temp arrays are?).  But then it should not be pnh(ilev) += exner(ilev), it needs to be pnh(ilev) = exner(ilev)?

Reply:
code should look like this:
```
m_elem_ops.compute_hydrostatic_p(kv,dp,p_i,pnh);
          Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
                               [&](const int ilev) {
            dp(ilev) = dp_adj(ilev);
          });
```

Reply:
Yeah, exner is used as a temporary holding place for hydrostatic pressure and will be recomputed to be exner later. But I am fairly confident that F90 and C++ match, and this is tested in the `theta-fhs*` tests. Here's the relevant code, in `prim_driver_base.F90`:
```
   do k=1,nlev
      phydro(:,:,k)=hvcoord%ps0*hvcoord%hyam(k) + ps(:,:)*hvcoord%hybm(k)
   enddo

   call pnh_and_exner_from_eos(...,pnh,...)
   do k=1,nlev
      pprime(:,:,k) = pnh(:,:,k)-phydro(:,:,k)
   enddo

   call get_hydro_pressure(phydro,...)
   do k=1,nlev
      pnh(:,:,k)=phydro(:,:,k) + pprime(:,:,k)
   enddo
```
and in `ForcingFunctor.hpp`:
```
   m_eos.compute_pnh_and_exner(...,pnh,...); // Same as call pnh_and_exner_from_eos(...,pnh,...)

   Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
                        [&](const int ilev) {
     pnh(ilev) -= m_hvcoord.ps0*m_hvcoord.hybrid_am(ilev) +
                            ps *m_hvcoord.hybrid_bm(ilev); // Same as pnh(:,:,k)-phydro(:,:,k)
     });

   m_elem_ops.compute_hydrostatic_p(...,exner); // call get_hydro_pressure(phydro,...)
   Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
                        [&](const int ilev) {
     pnh(ilev) += exner(ilev); // Same as pnh(:,:,k)=phydro(:,:,k) + pprime(:,:,k)
   });
```

Reply:
I verified that changing to just computing `m_elem_ops.compute_hydrostatic_p(kv,dp,p_i,pnh);` causes the cxx_vs_f90 tests to fail.

Reply:
thanks!  yes - my mistake - I completely misunderstood this function.